model/parch-key.js

```diff
@@ -0,0 +1,98 @@
+const TestResult = require('../utils/test-result.js');
+
+class ParchKey {
+
+    /**
+     * The name of the top level key object.
+     *
+     * @type {string}
+     */
+    name;
+    /**
+     * The id of the top level key object.
+     *
+     * @type {string}
+     */
+    id;
+
+    /**
+     * @class
+     *
+     * @since 1.0.0
+     *
+     * @param {string}  name    The name of the top level key object.
+     * @param {string}  id      The id of the top level key object.
+     */
+    constructor(name, id) {
+        // This class is abstract so make sure we cannot instantiate it
+        if (this.constructor === ParchKey) {
+            throw new TypeError('Abstract class "ParchKey" cannot be instantiated directly.');
+        }
+
+        this.name = name;
+        this.id = id;
+    }
+
+    /**
+     * Validate the name and id fields.
+     *
+     * Does not validate uniqueness.
+     *
+     * @since 1.0.0
+     *
+     * @returns {boolean} false if either name or id are empty, true otherwise.
+     */
+    validate() {
+        let valid = true;
+
+        if (ParchKey.testStringValue(this.name, 'name', 'ParchKey') !== TestResult.VALID) {
+            valid = false;
+        }
+        if (ParchKey.testStringValue(this.id, 'id', 'ParchKey') !== TestResult.VALID) {
+            valid = false;
+        }
```

Comment:
I left them separate in case I needed to do something different for each later, but I supposed I could just do that in the ```testStringValue``` function.

Reply:
They have been combined.